client/js/components/focus-trap.js

```diff
@@ -0,0 +1,47 @@
+import { addAttrToElements, removeAttrFromElements } from '../util';
+
+export default (el) => {
+  const trapStart = el.querySelector('.js-trap-start');
+  const trapEnd = el.querySelector('.js-trap-end');
+  const trapReverseStart = el.querySelector('.js-trap-reverse-start');
+  const trapReverseEnd = el.querySelector('.js-trap-reverse-end');
+
+  function focusTrapStart() {
+    trapStart.focus();
+  }
+
+  function focusTrapReverseStart({ shiftKey }) {
+    if (!shiftKey) return;
+
+    trapReverseStart.focus();
+  }
+
+  function addTrap() {
+    if (trapStart && trapEnd) {
+      addAttrToElements([trapStart, trapEnd], 'tabindex', '0');
+      trapEnd.addEventListener('focus', focusTrapStart);
+    }
+
+    if (trapReverseStart && trapReverseEnd) {
+      addAttrToElements([trapReverseStart, trapReverseEnd], 'tabindex', '0');
+      trapReverseEnd.addEventListener('keyup', focusTrapReverseStart);
+    }
+  }
+
+  function removeTrap() {
+    if (trapStart && trapEnd) {
+      removeAttrFromElements([trapStart, trapEnd], 'tabindex');
+      trapEnd.removeEventListener('focus', focusTrapStart);
+    }
+
+    if (trapReverseStart && trapReverseEnd) {
+      removeAttrFromElements([trapReverseStart, trapReverseEnd], 'tabindex');
+      trapReverseEnd.removeEventListener('keyup', focusTrapReverseStart);
+    }
+  }
+
+  return {
+    addTrap,
+    removeTrap
+  };
+};
```

Comment:
We need `fastdom` on this file, unless we have a better way or think it's not valid?

Reply:
This feels very complicated, and taking over browser behaviour, which, for people who rely on that, always feels a little tricky.

Could we not use grids and the DOM order to do this instead?

Reply:
> Could we not use grids and the DOM order to do this instead?

I don't believe so. Previously, for the mobile nav, we were `display: none`-ing everything else on the page (i.e. `.main` and `.footer`), but this obviously isn't a viable solution for anything that's going to live within the `.main` element. If we don't trap the focus, we allow for keyboard navigation to invisible elements (covered by a fullscreen overlay).

You're right about `fastdom` though – will update.

Reply:
Adding `fastdom` to the attr/class modifying utilities felt like a sensible approach.